File: CST-210/LockerPuzzle/LockerPuzzle/LockerSolver.cpp
#include "LockerSolver.h"

LockerSolver::LockerSolver(int numLockers)
{
    m_numberOfLockers = numLockers;
    m_lockerStatus = new bool[m_numberOfLockers];
    InitLockers();
}

void LockerSolver::InitLockers(void)
{
    for (auto lockerCnt = 0; lockerCnt < m_numberOfLockers; lockerCnt ++)
    {
        m_lockerStatus[lockerCnt] = false;
    }
}

int LockerSolver::GetNumberOfLockers(void)
{
    return m_numberOfLockers;
}

void LockerSolver::StudentLockerChange(int studentNum)
{
    auto lockerNum = studentNum - 1;
    while (lockerNum < m_numberOfLockers)
    {
        m_lockerStatus[lockerNum] = !m_lockerStatus[lockerNum];
        lockerNum += studentNum;
    }
}

bool *LockerSolver::GetLockerStatus(void)
{
    return m_lockerStatus;
}

File: CST-105/LockerPuzzle/src/LockerSolver.java

//
//  How to do the process:
//
//		for (lockerNumber form 1 to 100)  inclusive
//		{
//			locker = closed
//		}
//
//		for (studentNumber 1 to 100)  inclusive
//		{
//			start at locker corresponding to student
//			lockerNumber = studentNumber
//			while (locakerNumber < 100)
//			{
//				toggle locker status
//				add studentNumber to lockerNumber
//			}
//		}
//
//		find all the open lockers
//		for (lockNumber form 1 to 100) inclusive
//		{
//			if (lockerNumber is opened)
//			{
//				print lockerNumber
//			}
//		}
//

public class LockerSolver
{
	private int NumberOfLockers;
	private boolean[] LockerStatus;
	
	public LockerSolver(int numLockers)
	{
		NumberOfLockers = numLockers;
		LockerStatus = new boolean[NumberOfLockers];
	}
	
	public int GetNumberOfLockers()
	{
		return NumberOfLockers;
	}
	
	public void InitLockers()
	{
		for (var lockerCnt = 0; lockerCnt < NumberOfLockers; lockerCnt ++)
		{
			LockerStatus[lockerCnt] = false;
		}
	}
	
	public void StudentLockerChange(int studentNum)
	{
		var lockerNum = studentNum - 1;
		while (lockerNum < NumberOfLockers)
		{
			LockerStatus[lockerNum] = !LockerStatus[lockerNum];
			lockerNum += studentNum;
		}
	}
	
	public boolean[] GetLockerStatus()
	{
		return LockerStatus;
	}
}
